test_scripts/Defects/4_5/1912_Video_service_start_via_2_protocol.lua

```diff
@@ -0,0 +1,68 @@
+---------------------------------------------------------------------------------------------
+-- Issue: https://github.com/SmartDeviceLink/sdl_core/issues/1912
+-- Description
+-- SDL must respond NACK in case navigation app connected over protocol v2 sends StartService for audio service
+-- Preconditions
+-- SDL and HMI are started.
+-- Steps to reproduce
+-- navigation app connects over protocol v2 and this app sends StartService for audio service
+-- Actual result
+-- N/A
+-- Expected result
+-- SDL must respond StartService_NACK to this app
+---------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require("user_modules/script_runner")
+local constants = require('protocol_handler/ford_protocol_constants')
+local commonDefects = require('test_scripts/Defects/4_5/commonDefects')
+local events = require('events')
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+config.application1.registerAppInterfaceParams.appHMIType = {"NAVIGATION"}
+
+--[[ Local Functions ]]
+local function StartNaviServiceVia2Protocol(self)
+  self.mobileSession1:Send({
+      frameType = constants.FRAME_TYPE.CONTROL_FRAME,
+      serviceType = constants.SERVICE_TYPE.PCM,
+      frameInfo = constants.FRAME_INFO.START_SERVICE
+    })
+
+  local StartServiceNACK = events.Event()
+  StartServiceNACK.matches =
+  function(_, data)
+    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME and
+    data.serviceType == constants.SERVICE_TYPE.PCM and
+    data.sessionId == self.mobileSession1.sessionId and
+    data.frameInfo == constants.FRAME_INFO.START_SERVICE_NACK
+  end
+
+  local StartServiceACK = events.Event()
+  StartServiceACK.matches =
+  function(_, data)
+    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME and
+    data.serviceType == constants.SERVICE_TYPE.PCM and
+    data.sessionId == self.mobileSession1.sessionId and
+    data.frameInfo == constants.FRAME_INFO.START_SERVICE_ACK
+  end
+
+  self.mobileSession1:ExpectEvent(StartServiceNACK, "Expect StartServiceNACK")
+
+  self.mobileSession1:ExpectEvent(StartServiceACK, "Expect StartServiceACK")
```

Comment:
@GetmanetsIrina Taking in account peculiar execution of `Times(0)` I would propose to avoid to use it.
Please consider option with one expectation plus `ValidIf()`:
```
  local StartServiceNACKEvent = events.Event()
  StartServiceNACKEvent.matches =
  function(_, data)
    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME and
    data.serviceType == constants.SERVICE_TYPE.PCM and
    data.sessionId == self.mobileSession1.sessionId and
    (data.frameInfo == constants.FRAME_INFO.START_SERVICE_NACK or
      data.frameInfo == constants.FRAME_INFO.START_SERVICE_ACK)
  end

  self.mobileSession1:ExpectEvent(StartServiceNACKEvent, "Expect StartServiceNACK")
  :ValidIf(function(_, data)
      if data.frameInfo == constants.FRAME_INFO.START_SERVICE_NACK then
        return true
      else 
        return false, "StartService NACK received" 
      end
    end)
```